src/pipecat/transports/services/pipecat_cloud.py

```diff
@@ -0,0 +1,313 @@
+                params=daily_params,
+                input_name=input_name,
+                output_name=output_name,
+            )
+        elif isinstance(args, WebRTCSessionArguments):
+            logger.info("Using SmallWebRTCTransport")
+            transport_params = params.to_transport_params()
+            self._transport = SmallWebRTCTransport(
+                args.webrtc_connection,
+                transport_params,
+                input_name=input_name,
+                output_name=output_name,
+            )
+        else:
+            raise ValueError(f"Unsupported session arguments type: {type(args)}")
+
+        # Register any handlers that were added before transport creation
+        for event_name, handlers in self._pending_handlers.items():
+            for handler in handlers:
+                self._register_handler(event_name, handler)
+
+    def _register_handler(self, event_name: str, handler: Callable[..., Any]) -> None:
+        """Register a handler with the appropriate transport method."""
+        transport_type = type(self._transport)
+
+        # If the transport type has mappings and the event needs to be mapped
+        if (
+            transport_type in self._EVENT_MAPPINGS
+            and event_name in self._EVENT_MAPPINGS[transport_type]
+        ):
+            mapped_event = self._EVENT_MAPPINGS[transport_type][event_name]
+        else:
+            # Pass through the event name directly if no mapping exists
+            mapped_event = event_name
+
+        self._transport.event_handler(mapped_event)(handler)
+
+    def event_handler(self, event_name: str) -> Callable[..., Any]:
+        """Register an event handler.
+
+        Args:
+            event_name: The name of the event to handle. Common events:
+                - "on_client_connected": Called when a client connects
+                - "on_client_disconnected": Called when a client disconnects
+                Other event names are passed through to the underlying transport.
+
+        Returns:
+            A decorator that registers the handler function.
+        """
+
+        def decorator(handler: Callable[..., Any]) -> Callable[..., Any]:
+            if not hasattr(self, "_transport"):
+                # Store the handler to be registered when the transport is created
+                if event_name not in self._pending_handlers:
+                    self._pending_handlers[event_name] = []
+                self._pending_handlers[event_name].append(handler)
+            else:
+                self._register_handler(event_name, handler)
+            return handler
+
+        return decorator
+
+    async def start(self, frame):
+        """Start the transport."""
+        await self._transport.start(frame)
+
+    async def stop(self, frame):
+        """Stop the transport."""
+        await self._transport.stop(frame)
+
+    async def cancel(self, frame):
+        """Cancel the transport."""
+        await self._transport.cancel(frame)
```

Comment:
when are start/stop/cancel being used? this is usually added in frame processors, it shouldn't be needed here.